File: algorithms/python/sorting_algorithms/selection_sort.py
import  random
def selection_sort(_list):
    for i in range(len(_list)):
        minIndex = i
        for j in range(i+1, len(_list)):
            if (_list[minIndex] > _list[j]):
                minIndex = j
        _list[minIndex],_list[i]=_list[i],_list[minIndex]
    return _list

unsortedList = []
for i in range(10):
    unsortedList.append(random.randint(0,100))
print(f'The unsorted list: {unsortedList}')
result = selection_sort(unsortedList)
print(f'The sorted list: {result}')

File: algorithms/java/sorting_algorithms/selection_sort.java
package sorting_algorithms;
import java.util.ArrayList;
public class selection_sort {
	public static void main(String args[]) {
		
		ArrayList unsortedList = new ArrayList();
		for(int i = 0 ; i < 10 ;i++) {
			unsortedList.add(createRondomNum(100));
		}
		System.out.println("The unsorted list : "+unsortedList );
		ArrayList result = selection_sort(unsortedList);
		System.out.println("The sorted list : "+result);
	}
		
	private static ArrayList selection_sort(ArrayList list) {
		for (int i = 0; i < list.size();i++) {
			int minIndex = i;
			for (int j = i+1;j<list.size();j++) {
				if( (int) list.get(minIndex) > (int) list.get(j) ) {
					minIndex = j;
				}		
			}
			int temp = (int) list.get(minIndex);
			list.set(minIndex, list.get(i));
			list.set(i, temp);
		}
		return list;
	}


	private static int createRondomNum(int i) {
		int number = (int) ( i*Math.random()) ;
		return number;
	}

}
